translators/__tests__/ChatGPTTranslator.test.js

```diff
@@ -0,0 +1,79 @@
+import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const translatorCode = fs.readFileSync(
+    path.resolve(__dirname, '../ChatGPTTranslator.js'),
+    'utf8'
+);
+
+const ChatGPTTranslator = eval(`${translatorCode}`);
+
+describe('ChatGPTTranslator', () => {
+    let translator;
+    
+    beforeAll(() => {
+        translator = new ChatGPTTranslator('gpt-4', process.env.OPEN_API_KEY);
+    });
+
+    test('translates text from English to Spanish', async () => {
+        const text = 'Hello, world!';
+        const result = await translator.translate(text, 'en', 'es');
+        expect(result).toBeTruthy();
+        expect(typeof result).toBe('string');
+    });
+
+    test('translates batch of texts', async () => {
+        const texts = ['Hello', 'Good morning'];
+        const results = await translator.translateBatch(texts, 'en', 'fr');
+        expect(Array.isArray(results)).toBe(true);
+        expect(results.length).toBe(2);
+        expect(results.every(r => typeof r === 'string')).toBe(true);
+    });
+
+    test('handles empty input', async () => {
+        const result = await translator.translate('', 'en', 'es');
+        expect(result).toBe('');
+    });
+
+    test('handles length limit correctly', () => {
+        const longText = 'a'.repeat(21000);
+        expect(translator.checkLimitExceeding(longText)).toBe(1);
+        expect(translator.checkLimitExceeding('short text')).toBe(0);
+    });
```

Comment:
This method must return the number of chars that is out of limit. It is necessary for Linguist batch scheduler to efficiently group texts by its size.

You may check example here https://github.com/translate-tools/linguist-translators/blob/4e9c69cc7d2ea3ed68c0d015a3357565a10c352a/translators/LibreTranslator.js#L45-L51

I think in this test a first assert should expect 1000 (that means we have to cut at least 1k chars) and second one is -19990 (that means we can add up to 19990 characters, and still fit in single request)